### PR TITLE
DEV2-3003: Change force registeration flow

### PR DESCRIPTION
**Pull request by ofekby**

### CODE REVIEW

File: Common/src/main/java/com/tabnineCommon/binary/requests/capabilities/CapabilitiesResponse.kt [hunk 1]
```diff
@@ -7,4 +7,22 @@ import com.tabnineCommon.capabilities.Capability
...
 ) : BinaryResponse
+
+enum class ExperimentSource {
+    @SerializedName("API")
+    API,
+
+    @SerializedName("APIErrorResponse")
+    APIErrorResponse,
+
+    @SerializedName("Hardcoded")
```
Comment (assapir) [inline Common/src/main/java/com/tabnineCommon/binary/requests/capabilities/CapabilitiesResponse.kt:21]: I think you don't need the attribute if it the same

File: Tabnine/src/main/java/com/tabnine/statusBar/TabnineIconProvider.kt [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    @JvmStatic
+    fun getIcon(
+        serviceLevel: ServiceLevel?,
+        isLoggedIn: Boolean?,
+        cloudConnectionHealthStatus: CloudConnectionHealthStatus
+    ): Icon {
+        if (!CapabilitiesService.getInstance().isReady || isLoggedIn == null || serviceLevel == null || (
+            CapabilitiesService.getInstance()
+                .isCapabilityEnabled(Capability.FORCE_REGISTRATION) &&
+                !isLoggedIn && serviceLevel == ServiceLevel.FREE
```
Comment (assapir) [inline Tabnine/src/main/java/com/tabnine/statusBar/TabnineIconProvider.kt:21]: if the user isn't logged in they are free, always, no?

**Review by assapir — APPROVED** [review]

File: Common/src/main/java/com/tabnineCommon/binary/requests/capabilities/CapabilitiesResponse.kt (hunk 1 above)
Comment (ofekby) [reply Common/src/main/java/com/tabnineCommon/binary/requests/capabilities/CapabilitiesResponse.kt:21]: I prefer keeping, just to be explicit, enums are weird :}

File: Tabnine/src/main/java/com/tabnine/statusBar/TabnineIconProvider.kt (hunk 2 above)
Comment (ofekby) [reply Tabnine/src/main/java/com/tabnine/statusBar/TabnineIconProvider.kt:21]: Yes, not logged in and Free meaning they should not have the starter icon, just Tabnine. (Same as vscode)

File: Tabnine/src/test/java/com/tabnine/healthCheck/StatusBarWidgetTests.java [hunk 3]
```diff
@@ -3,25 +3,45 @@
...
     TabnineStatusBarWidget widget = new TabnineStatusBarWidget(myFixture.getProject());
-    HealthCheckTestUtils.notifyHealthStatus(CloudConnectionHealthStatus.Ok);
+    HealthCheckTestUtils.notifyStateForWidget(
+        ServiceLevel.FREE, true, CloudConnectionHealthStatus.Ok);
+    MockBinaryResponse.mockCapabilities(binaryProcessGatewayMock, "API", new ArrayList<>());
+    CapabilitiesService.getInstance().init();
 
-    assertEquals(widget.getIcon(), StaticConfig.ICON_AND_NAME_STARTER);
+    while (!CapabilitiesService.getInstance().isReady()) {
+      Thread.sleep(100);
```
Comment (yanir-codota) [inline Tabnine/src/test/java/com/tabnine/healthCheck/StatusBarWidgetTests.java:26]: why?

**Review by yanir-codota — APPROVED** [review]

File: Tabnine/src/test/java/com/tabnine/healthCheck/StatusBarWidgetTests.java (hunk 3 above)
Comment (ofekby) [reply Tabnine/src/test/java/com/tabnine/healthCheck/StatusBarWidgetTests.java:26]: It is an asynchronous test

**Pull request merged**